<launch>
<!-- add camera config file here -->
  <arg name="_camera_info_url" value="file:///home/advaith/.ros/camera_info/cam1.yaml"/>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
  <!-- change the video device if not using built-in webcam -->
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="camera" />
    <param name="io_method" value="mmap"/>
  </node>
  <node name="image_proc" pkg="image_proc" type="image_proc" ns="usb_cam"/>
  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node>
  <!-- transforms -->
  <node pkg="tf" type="static_transform_publisher" name="tf_transform" args="1 0 0 0 0 0 1 /camera /saifer_tool 100" />

  <arg name="launch_prefix" default="" /> 
  <arg name="node_namespace" default="tool_pose" />
  <arg name="camera_name" default="usb_cam" />
  <arg name="camera_frame" default="saifer_tool" />
  <arg name="image_topic" default="usb_cam" />

  <!-- Set parameters -->
  <rosparam command="load" file="$(find visualize)/config/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find visualize)/config/tags.yaml" ns="$(arg node_namespace)" />
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="/usb_cam/image_rect" />
    <remap from="camera_info" to="/usb_cam/camera_info" />

    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>

 </launch>